src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -690,61 +686,62 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 			}
 		}
 
-		// Convert to typed editors and options
-		const typedEditors: IEditorInputWithOptions[] = editors.map(editor => {
-			if (isEditorInputWithOptions(editor)) {
-				return editor;
-			}
-
-			return {
-				editor: this.createEditorInput(editor),
-				options: editor.options
-			};
-		});
-
 		// Find target groups to open
-		const mapGroupToEditorsCandidates = new Map<IEditorGroup, IEditorInputWithOptions[]>();
+		const mapGroupToEditorsCandidates = new Map<IEditorGroup, Array<IEditorInputWithOptions | IUntypedEditorInput>>();
 		if (group === SIDE_GROUP) {
-			mapGroupToEditorsCandidates.set(this.findSideBySideGroup(), typedEditors);
+			mapGroupToEditorsCandidates.set(this.findSideBySideGroup(), editors);
 		} else {
-			for (const typedEditor of typedEditors) {
-				const targetGroup = this.findTargetGroup(typedEditor.editor, typedEditor.options, group);
+			for (const editor of editors) {
+
```

Comment:
I prefer statements without newline within `for` unless you have a comment above